dunai-frp-bearriver/src/FRP/BearRiver.hs

```diff
@@ -436,6 +437,12 @@ dSwitch sf sfC = MSF $ \a -> do
                        return (b, ct')
     (b, NoEvent) -> return (b, dSwitch ct sfC)
 
+link :: Monad m => SF m a (Either b c) -> (c -> SF m a (Either b d)) -> SF m a (Either b d)
+link sf sfC = MSF $ \a -> do
```

Comment:
It's always a maintenance effort to add new primitives that use the `MSF` constructor. Have you tried to implement this using exceptions (https://hackage.haskell.org/package/dunai-0.5.2.1/docs/Control-Monad-Trans-MSF-Except.html)?

Reply:
I have: you can see my experiment in [one of the versions in this branch][except]. It uses the output type `(b, Event c)` instead of `Either b c`, but that's probably easy to change.

However, it's not obvious to me how to ensure that the time step is zero for the actions after the switching event. This version is simpler and closely matches the implementations of `switch` and `dSwitch`.

[except]: https://github.com/ivanperez-keera/dunai/pull/242/commits/07623ed1e0ed4ab2374cdeff1297e1365a38eac2#diff-9c2aba48d4d293c9782080f4347a9074e906e6476cece93c907958e8282c1a06R276